src/commands/azureServiceOperators/ui/azureservicebrowser.ts

```diff
@@ -0,0 +1,117 @@
+    }
+}
+
+ class AzureServicesFolderNode implements k8s.ClusterExplorerV1.Node {
+    constructor(private readonly explorer: k8s.ClusterExplorerV1) {}
+    async getChildren(): Promise<k8s.ClusterExplorerV1.Node[]> {
+        const pinnedNodes = (await pinnedServiceKinds()).map((k) => new PinnedAzureServiceKindNode(this.explorer, k));
+        const allNode = new AllAzureServicesFolderNode(this.explorer);
+        return [...pinnedNodes, allNode];
+    }
+    getTreeItem(): TreeItem {
+        const treeItem = new TreeItem("Azure Services", TreeItemCollapsibleState.Collapsed);
+        treeItem.contextValue = "aks.aso.azureservices";
+        return treeItem;
+    }
+}
+
+ class AllAzureServicesFolderNode implements k8s.ClusterExplorerV1.Node {
+    constructor(private readonly explorer: k8s.ClusterExplorerV1) {}
+    async getChildren(): Promise<k8s.ClusterExplorerV1.Node[]> {
+        const all = await allServiceKinds();
+
+        return all!.map((k) => new AzureServiceKindNode(this.explorer, k));
+    }
+    getTreeItem(): TreeItem {
+        const treeItem = new TreeItem("All Service Types", TreeItemCollapsibleState.Collapsed);
+        treeItem.contextValue = "aks.aso.allservices";
+        return treeItem;
+    }
+
+ }
+
+ class AzureServiceKindNode  implements k8s.ClusterExplorerV1.Node {
+    constructor(private readonly explorer: k8s.ClusterExplorerV1, private readonly kind: AzureServiceKind) {}
+    async getChildren(): Promise<k8s.ClusterExplorerV1.Node[]> {
+        const allFolderChildren = this.explorer.nodeSources.resourceFolder(this.kind.displayName, this.kind.displayName, this.kind.manifestKind, this.kind.abbreviation);
+        return allFolderChildren.nodes();
+    }
+    getTreeItem(): TreeItem {
+        const treeItem = new TreeItem(this.kind.displayName, TreeItemCollapsibleState.Collapsed);
+        treeItem.contextValue = "aks.aso.allservices.serviceitem";
+
+        return treeItem;
+    }
+}
+
+class PinnedAzureServiceKindNode  implements k8s.ClusterExplorerV1.Node {
+    constructor(private readonly explorer: k8s.ClusterExplorerV1, private readonly kind: AzureServiceKind) {}
+    async getChildren(): Promise<k8s.ClusterExplorerV1.Node[]> {
+        const allFolderChildren = this.explorer.nodeSources.resourceFolder(this.kind.displayName, this.kind.displayName, this.kind.manifestKind, this.kind.abbreviation);
+
+        return allFolderChildren.nodes();
+    }
+    getTreeItem(): TreeItem {
+        const treeItem = new TreeItem(this.kind.displayName, TreeItemCollapsibleState.Collapsed);
+        treeItem.contextValue = "aks.aso.azureservices.pinnedserviceitem";
+        return treeItem;
+    }
+}
+
+export interface AzureServiceKind {
+    readonly displayName: string;
+    readonly manifestKind: string;
+    readonly abbreviation: string;
+}
+
+async function pinnedServiceKinds(): Promise<AzureServiceKind[]> {
+    const pinnedKindNames = <AzureServiceKind[]>vscode.workspace.getConfiguration().get('aso.pinned');
```

Comment:
I'm puzzled by this type assertion.  Your `package.json` says that `aso.pinned` is an array of strings, not an array of structured objects.

Reply:
I mean I am all in favour of being able to override the display name in the pinned folder - it just didn't look like the configuration schema supported it.